components/txn_types/src/write.rs

```diff
@@ -117,6 +117,18 @@ pub struct Write {
     /// For CDC and TiFlash, when they receives a commit record with `gc_fence` field set, it can
     /// determine that it must be caused by an overlapped rollback instead of an actual commit.
     ///
+    /// Note: GC fence will only be written on `PUT` and `DELETE` versions, and may only point to
```

Comment:
It seems to me that a `DELETE` doesn't need a fence. If there is a newer `PUT`, the `PUT` will not be GCed.

Reply:
Oh, if TiFlash needs the fence to distinguish a rewritten write record, it's reasonable.